### PR TITLE
Grab pay icon

### PR DESCRIPTION
**Pull request by jawad-eGHL**
** Checklist **
- [ * ] All icons have a corresponding entry in `db/payment_icons.yml`
- [ * ] I have followed the icon guidelines detailed in the [CONTRIBUTING.md](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md) file
- [ * ] I have optimized the icon with [SVGO](https://jakearchibald.github.io/svgomg/)
- [ * ] I am confident that all icons are clear and easy to read/understand
- [ * ] I have provided a link to the brand icon’s brand guidelines whenever possible.

If this pull request is not adding new icons, you can remove this checklist.

### CODE REVIEW

File: app/assets/images/payment_icons/grabpay.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...3c3uMhZB6ogT00fGHr9CEhGcK1qmKuB6jUZ5n/Gyu24c7fBdzcewre0G wutNjhjXfNmhfJWxEOK11n1kLEptvsSKYuJMB2wHP6lV4D5jFIm7FcII4liy9TXGMxraWjjj2RI8 +BgkPSLH1aBdxlyVpMB8ZCxa8eNolVfLhAVPI+ybvMzjPuOeRQMamnkBsYJbRVVfYQx3004yYvXg kmVHfbyfsfircb+drrl2lzE/1pQKpg3GQrsYmo1O6AB3jJChu+D4Ydsl+1rNzgJGRU8cjTFuLkfW 7FzKUw9mHxyl+NllzEVJ28BpjC31iYH+DDxImWMFvFlqi4Jm2mMM+36wvLafMD5jQnLZKYHVS6YM FzZTg1ROfXmNpd/zDcaeTNh+eMEDq1460aRIHQVlRyzffsmY2UjzqT8vRW4Wafl4NYH+QrwlfWbw 2G2GTxe9t6/EccpD0xf5WWbsy0XoRl9oTnMEe6qVblXQK2GyyMQzxnhG5uRQw/ZRUUTB4lQaY3Zm jNVxwJuxs3blCHWX8frwsYWVcz3pZHVrZgD9Rj9IOfWgD/9czN1ctHh05uNpa3Z8xZh1noonIZZL thjjoY9jS81G9Ci11P6FfOyUWSKFksZY0q7tE7wWvWAoFBk/xdh9hUY+Z7wcnZXT2J7PUadp9Uo5 580i2mxkAVlqm9RfqrkE9Q3GknY9eUwfLi2S+HY+9ffzbWzYBJYji/sXjK2L+uRrmzGlk+q2VrrD 7TMmk97J4brFWGhX+ezdKOI92AMQj3hfMaaaJ+/NnzG2/FEZI3s8hfjCToJoT9pivplSGfv899x6 3lWhv95U5msZ/JIxvbk+ZIvX3sgzxnCinrO5KZPuvtHNFxjTAk4K36EVD+JUxmfe4vULM6dxEm/s JXEr+0OIx9an+/35+yckOJ2YEPSvGtNm/bNOjvx07eVbLqSOE1ozpFmOgu2zW9TtG/eEwdrF8XSp 7/3PvxD5s+AbY+XJ5V+NjedPfzfuQnn/tCW/CbXICfX3O/t8+O2av7Lvd/fxOCuv+f3MW8qfBeXE +/D9/j4fMuPqjf/z4XMhMZ5+7n8bfBIE43L+X6ywYJyN7vf7+m8gKock9/rvd2RgYGBgYGBgYGBg YGBgYGBgYGBgYGBgYGBgYGBg8BfjH4/BoK/HnP8JAAAAAElFTkSuQmCC"/></svg>
```
Comment (tauthomas01) [inline app/assets/images/payment_icons/grabpay.svg]: There's a few things that need some changes:

- `id="pi-Layer_1"` attributes is unused. Can we remove it?

[alt](https://screenshot.click/18-56-dk2oe-p5mmo.png)

- we need a border around the icon. See the example in the [contribution](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md#appearance) document.

Comment (jawad-eGHL) [reply app/assets/images/payment_icons/grabpay.svg]: Hi @tauthomas01 ,
thanks for the feedback, ill do the changes and get back to you.

File: app/assets/images/payment_icons/grabpay.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...XGMxraWjjj2RI8 +BgkPSLH1aBdxlyVpMB8ZCxa8eNolVfLhAVPI+ybvMzjPuOeRQMamnkBsYJbRVVfYQx3004yYvXg kmVHfbyfsfircb+drrl2lzE/1pQKpg3GQrsYmo1O6AB3jJChu+D4Ydsl+1rNzgJGRU8cjTFuLkfW 7FzKUw9mHxyl+NllzEVJ28BpjC31iYH+DDxImWMFvFlqi4Jm2mMM+36wvLafMD5jQnLZKYHVS6YM FzZTg1ROfXmNpd/zDcaeTNh+eMEDq1460aRIHQVlRyzffsmY2UjzqT8vRW4Wafl4NYH+QrwlfWbw 2G2GTxe9t6/EccpD0xf5WWbsy0XoRl9oTnMEe6qVblXQK2GyyMQzxnhG5uRQw/ZRUUTB4lQaY3Zm jNVxwJuxs3blCHWX8frwsYWVcz3pZHVrZgD9Rj9IOfWgD/9czN1ctHh05uNpa3Z8xZh1noonIZZL thjjoY9jS81G9Ci11P6FfOyUWSKFksZY0q7tE7wWvWAoFBk/xdh9hUY+Z7wcnZXT2J7PUadp9Uo5 580i2mxkAVlqm9RfqrkE9Q3GknY9eUwfLi2S+HY+9ffzbWzYBJYji/sXjK2L+uRrmzGlk+q2VrrD 7TMmk97J4brFWGhX+ezdKOI92AMQj3hfMaaaJ+/NnzG2/FEZI3s8hfjCToJoT9pivplSGfv899x6 3lWhv95U5msZ/JIxvbk+ZIvX3sgzxnCinrO5KZPuvtHNFxjTAk4K36EVD+JUxmfe4vULM6dxEm/s JXEr+0OIx9an+/35+yckOJ2YEPSvGtNm/bNOjvx07eVbLqSOE1ozpFmOgu2zW9TtG/eEwdrF8XSp 7/3PvxD5s+AbY+XJ5V+NjedPfzfuQnn/tCW/CbXICfX3O/t8+O2av7Lvd/fxOCuv+f3MW8qfBeXE +/D9/j4fMuPqjf/z4XMhMZ5+7n8bfBIE43L+X6ywYJyN7vf7+m8gKock9/rvd2RgYGBgYGBgYGBg YGBgYGBgYGBgYGBgYGBgYGBg8BfjH4/BoK/HnP8JAAAAAElFTkSuQmCC"/><rect x="0" y="0" width="38" height="24" rx="2" ry="2" fill="none" stroke="#000" stroke-width="1" stroke-opacity=".7"/></svg>
```
Comment (tauthomas01) [inline app/assets/images/payment_icons/grabpay.svg]: - The border doesn't look the same as other icon. See this [video demonstration](https://screenshot.click/20-48-ivxf5-zwa2m.mp4). You can add this code in your SVG. For more info, see the [contribution](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md#template) document.

```svg
<path fill="#000" opacity=".07" d="M35 0H3C1.3 0 0 1.3 0 3v18c0 1.7 1.4 3 3 3h32c1.7 0 3-1.3 3-3V3c0-1.7-1.4-3-3-3z"/>
<path fill="#fff" d="M35 1c1.1 0 2 .9 2 2v18c0 1.1-.9 2-2 2H3c-1.1 0-2-.9-2-2V3c0-1.1.9-2 2-2h32"/>
```

- The payment icon looks big. Is it possible to resize?

[alt](https://screenshot.click/20-53-7bzsw-qgxaa.png)

**Comment by jawad-eGHL** [comment]
I wonder why all checks failed? Though the icon is adjusted.

**Comment by jawad-eGHL** [comment]
@tauthomas01 what does the following build error indicates

> PaymentIconTest#test_Payment_icon_SVGs_with__or_<image>_tags_base64-encode_PNG_image_data:
NoMethodError: undefined method `truncate' for nil:NilClass

**Comment by tauthomas01** [comment]
@jawad-eGHL I'm unsure about the error you mentioned. If we try to fix other errors, maybe this error will get fixed. Can you try to fix the following `The 'grabpay' SVG file should have a <title> tag as first child of the root <svg> tag`?

You can add back the `<title>` tag after the opening of the SVG tag:
```
<title id="pi-grabpay">GrabPay</title>
```

**Comment by jawad-eGHL** [comment]
Hi @tauthomas01 
the reason for build failure is still the same and unknown to me.

**Comment by tauthomas01** [comment]
Hi @jawad-eGHL ,

I think I found the issue: if your icon use "base64" type image, it needs to use `xlink:href` attribute. Your icon is using `href` and the tests are looking for the `xlink:href` attribute. That's why it fails.

After looking at the icon, it looks pixelated so if you zoom in at 500%, it does not keep the same quality. Can you please change into a vector format?
[alt](https://screenshot.click/03-16-ul0yb-3cvdc.png)

**Comment by jawad-eGHL** [comment]
Hi @tauthomas01 
image
at 500% magnification

File: app/assets/images/payment_icons/grabpay.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...11.034 9.743-.897.116-2.108.356-2.693.532-.915.277-1.227.262-2.257-.107-.657-.235-1.59-.426-2.072-.424-3.564.015-11.286-6.86-11.286-10.049 0-.261-.18-.926-.4-1.477-.455-1.135-.139-1.923 1.337-3.335.984-.941 1.287-.657 2.605 2.439 4.758 11.176 15.563 12.035 22.168 1.762.442-.687.484-1.253.486-6.52l.002-5.768-1.078-2.132c-3.251-6.43-11.791-8.228-17.979-3.784-1.945 1.397-10.455 10.201-11.469 11.865-.519.852-2.268 2.951-3.887 4.666-2.991 3.167-4.069 4.542-4.695 5.99-.236.548-.575.856-1.054.962-.389.085-.838.195-.999.243-.44.133-.585-.482-.585-2.48 0-2.465.403-3.16 4.077-7.021 2.807-2.95 4.446-4.978 4.448-5.505.002-.278 7.64-8.108 10.66-10.926.414-.386 1.26-.981 1.881-1.322l1.63-.896c4.065-2.247 7.551-2.879 10.745-1.948m111.386 22.927c-1.103.08-2.232.274-2.507.43-.276.156-1.518.473-2.759.706-4.878.912-8.401 3.864-8.401 7.038 0 1.858 2.627 5.832 4.298 6.501 2.517 1.007 11.25-.034 11.25-1.341 0-.088.393-.372.873-.63.48-.259.937-.673 1.016-.922.079-.249.335-.514.569-.59 1.168-.378 1.555-1.88 1.555-6.029 0-5.727.139-5.604-5.894-5.163" fill="#fff"/><rect stroke-opacity=".07" stroke="#000" fill="none" ry="21.8" rx="21.8" height="240.347" width="389.473" y="6" x="6" stroke-width="10.9"/></svg>
```
Comment (tauthomas01) [inline app/assets/images/payment_icons/grabpay.svg]: - In the [contribution](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md#markup) document, `viewBox` attribute needs to have `viewBox="0 0 38 24"`.
- There is a small difference in the border of the icon. Can you modify it? Here's what is different compared to other icon: [video demonstration](https://screenshot.click/07-45-0it1n-gr2gv.mp4)

**Comment by jawad-eGHL** [comment]
@tauthomas01 
Hope, it fully applies the specifications

**Pull request merged**